### PR TITLE
Refactor `databricks_recipient` to Go SDK

### PR DESCRIPTION
**Pull request by nkvuong**
## Changes

- Refactor `databricks_recipient` to Go SDK
- Partially solve #2587 (need to refactor `databricks_share` as well)

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [x] covered with integration tests in `internal/acceptance`
- [ ] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Review by alexott — COMMENTED** [review]
code-wise, lgtm, few comments about docs

File: docs/resources/recipient.md [hunk 1]
```diff
@@ -99,11 +100,7 @@ In addition to all arguments above, the following attributes are exported:
   * `expiration_time` - Expiration timestamp of the token in epoch milliseconds.
   * `updated_at` - Time at which this recipient Token was updated, in epoch milliseconds.
   * `updated_by` - Username of recipient Token updater.
-
-## Attribute Reference
-
-In addition to all arguments above, the following attributes are exported:
-
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply docs/resources/recipient.md:106]: it was duplicated from the same block above

File: docs/resources/recipient.md [hunk 2]
```diff
@@ -67,13 +67,14 @@ The following arguments are required:
...
 * `sharing_code` - (Optional) The one-time sharing code provided by the data recipient.
+* `owner` - (Optional) Username/groupname/sp application_id of the recipient owner.
 * `authentication_type` - (Optional) The delta sharing authentication type. Valid values are `TOKEN` and `DATABRICKS`.
 * `data_recipient_global_metastore_id` - Required when authentication_type is DATABRICKS.
 * `ip_access_list` - (Optional) The one-time sharing code provided by the data recipient.
 
 ### Ip Access List Argument
 
-Only one `ip_access_list` blocks is allowed in a recipient. It conflicts with authentication type DATABRICKS.
+Only one `ip_access_list` blocks is allowed in a recipient. It conflicts with authentication type `DATABRICKS`.
```
[the remark this answers is not in the document]
Comment (nkvuong) [reply docs/resources/recipient.md:77]: done

**Review by alexott — APPROVED** [review]
lgtm

**Review by mgyucht — APPROVED** [review]
Nice, this is a great step in the right direction. Over time, we'll have to migrate the StructToSchema argument as well.

**Pull request merged**